File: challenging/2_longest_string_chain/longest_string_chain_test.py
import unittest
from longest_string_chain import longestStringChain

class TestProgram(unittest.TestCase):
    def test_case_1(self):
        strings = ["abde", "abc", "abd", "abcde", "ade", "ae", "1abde", "abcdef"]
        expected = ["abcdef", "abcde", "abde", "ade", "ae"]
        self.assertEqual(longestStringChain(strings), expected)

File: challenging/2_longest_string_chain/longest_string_chain.test.js
const longestStringChain = require('./longest_string_chain');

describe('TestProgram', () => {
  it('Test Case 1', () => {
    const strings = ['abde', 'abc', 'abd', 'abcde', 'ade', 'ae', '1abde', 'abcdef'];
    const expected = ['abcdef', 'abcde', 'abde', 'ade', 'ae'];
    expect(longestStringChain(strings)).toEqual(expected);
  });
});
